### PR TITLE
feat(testing): use kustomize to patch deployments before deploy

### PR DESCRIPTION
**Pull request by CarterFendley**
### Description of your changes:

This PR is a followup to #11284 which removes the `kubectl patch` commands by introducing a kustomization. This is in attempt to solve #11293, although does not yet address the Tekton side of things.

**Feedback wanted:**
- Should this new `kustomization.yaml` be moved to [manifests/kustomize/env](https://github.com/kubeflow/pipelines/tree/master/manifests/kustomize/env)?

### Checklist:
- [x] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: scripts/deploy/github/deploy-kfp.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ then
   exit $EXIT_CODE
 fi
 
-PLATFORM_AGNOSTIC_MANIFESTS="manifests/kustomize/env/platform-agnostic"
+PLATFORM_AGNOSTIC_MANIFESTS="scripts/deploy/github/manifests/test"
```
Comment (hbelmiro) [inline scripts/deploy/github/deploy-kfp.sh:35]: Can we rename `PLATFORM_AGNOSTIC_MANIFESTS` to `TEST_MANIFESTS`?

**Review by hbelmiro — COMMENTED** [review]
/ok-to-test

Thank you for working on this @CarterFendley.
Looks good. I just left a small suggestion to rename a variable.

> Feedback wanted:
> 
> Should this new kustomization.yaml be moved to [manifests/kustomize/env](https://github.com/kubeflow/pipelines/tree/master/manifests/kustomize/env)?

I don't have a strong opinion. In case we decide to move, we should move it to a new `test` directory and include this directory in the [paths that trigger tests](https://github.com/kubeflow/pipelines/blob/4ccb047771f6dae5d612d176d0e23c2fbb97e6a1/.github/workflows/e2e-test.yml#L8). If we leave them where they are, we don't need to change anything else?

@HumairAK @DharmitD thoughts?

File: scripts/deploy/github/deploy-kfp.sh (hunk 1 above)
Comment (CarterFendley) [reply scripts/deploy/github/deploy-kfp.sh:35]: Ah yes, good point, will do!

File: scripts/deploy/github/manifests/test/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  newTag: latest
+- name: gcr.io/ml-pipeline/persistenceagent
+  newName: kind-registry:5000/persistenceagent
+  newTag: latest
+- name: gcr.io/ml-pipeline/scheduledworkflow
+  newName: kind-registry:5000/scheduledworkflow
+  newTag: latest
+
+patchesStrategicMerge:
+- overlays/apiserver-env.yaml
```
Comment (DharmitD) [inline scripts/deploy/github/manifests/test/kustomization.yaml:19]: Make sure to fix EoF here.

File: scripts/deploy/github/manifests/test/overlays/apiserver-env.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+spec:
+  template:
+    spec:
+      containers:
+        - name: ml-pipeline-api-server
+          env:
+            - name: V2_DRIVER_IMAGE
+              value: kind-registry:5000/driver
+            - name: V2_LAUNCHER_IMAGE
+              value: kind-registry:5000/launcher
```
Comment (DharmitD) [inline scripts/deploy/github/manifests/test/overlays/apiserver-env.yaml:14]: Make sure to fix EoF here.

**Comment by DharmitD** [comment]
> I don't have a strong opinion. In case we decide to move, we should move it to a new `test` directory and include this directory in the [paths that trigger tests](https://github.com/kubeflow/pipelines/blob/4ccb047771f6dae5d612d176d0e23c2fbb97e6a1/.github/workflows/e2e-test.yml#L8). If we leave them where they are, we don't need to change anything else?

+1, no need to move as such, but if we decide to move it, make sure to create a new `test` directory and also move the `overlays` directory too.

**Comment by CarterFendley** [comment]
The recent additions to the PR move the contents of `scripts/deploy/github` to `.github/resources/` manifests have been seperated in to two different directories. This structure was suggested [in slack here](https://cloud-native.slack.com/archives/C073N7BMLB1/p1729022202731139?thread_ts=1728481551.330219&cid=C073N7BMLB1).

- `.github/resources/manifests/argo`
- `.github/resources/manifests/tekton`

There is some similarity between the Argo testing kustomization and the Tekton ones, but as per discussion in that slack thread, tekton will be tracked by #11292 and not handled in this PR.

**Comment by hbelmiro** [comment]
@CarterFendley can you please rebase?

**Comment by rimolive** [comment]
/lgtm

The tests passed in the new paths

**Review by DharmitD — APPROVED** [review]
/Approve

**Pull request merged**